src/zcbor_decode.c

```diff
@@ -321,19 +329,108 @@ bool zcbor_bstr_end_decode(zcbor_state_t *state)
 }
 
 
+static void partition_fragment(zcbor_state_t *state,
+	struct zcbor_string_fragment *result)
+{
+	result->fragment.len = MIN(result->fragment.len,
+		(size_t)state->payload_end - (size_t)state->payload);
+}
+
+
+static bool start_decode_fragment(zcbor_state_t *state,
+	struct zcbor_string_fragment *result,
+	zcbor_major_type_t exp_major_type)
+{
+	if(!str_start_decode(state, &result->fragment, exp_major_type)) {
+		ZCBOR_FAIL();
+	}
+
+	result->offset = 0;
+	result->total_len = result->fragment.len;
+	partition_fragment(state, result);
+	state->payload_end = state->payload + result->fragment.len;
+
+	return true;
+}
+
+bool zcbor_bstr_start_decode_fragment(zcbor_state_t *state,
+	struct zcbor_string_fragment *result)
+{
+	if (!start_decode_fragment(state, result, ZCBOR_MAJOR_TYPE_BSTR)) {
+		ZCBOR_FAIL();
+	}
+	if (!zcbor_new_backup(state, ZCBOR_MAX_ELEM_COUNT)) {
+		FAIL_RESTORE();
+	}
+	return true;
+}
+
+
+void zcbor_next_fragment(zcbor_state_t *state,
+	struct zcbor_string_fragment *prev_fragment,
+	struct zcbor_string_fragment *result)
+{
+	memcpy(result, prev_fragment, sizeof(*result));
+	result->fragment.value = state->payload_mut;
```

Comment:
Why is this here?

Reply:
The fragment is assumed to continue at the payload, since the user is assumed to have called `zcbor_update_state()` prior to calling this. I added some function docs explaining this.